src/guide/advanced/permissions.md

```diff
@@ -0,0 +1,286 @@
+let permission_token_to_alice: PermissionToken =
+    transfer::CanTransferUserAssets::new(alice_xor_id).into();
+
+// Create an instruction that grants Bob permission to transfer `alice_xor_id` asset
+let grant = Instruction::Grant(GrantBox::new(
+    permission_token_to_alice,
+    IdBox::AccountId(bob_id.clone()),
+));
+```
+
+### `CanTransferOnlyFixedNumberOfTimesPerPeriod`
+
+With `CanTransferOnlyFixedNumberOfTimesPerPeriod` permission token, a user
+can transfer assets only a fixed number of times per some time period.
+
+```rust
+const PERIOD_MS: u64 = 5000;
+const COUNT: u32 = 2;
+
+// Create a new `CanTransferOnlyFixedNumberOfTimesPerPeriod` permission token
+// that limits the number of transfer to 2 per 5000 ms.
+let permission_token =
+    transfer::CanTransferOnlyFixedNumberOfTimesPerPeriod::new(PERIOD_MS.into(), COUNT);
+```
+
+### `CanSetKeyValueInUserAssets`
+
+With `CanSetKeyValueInUserAssets` permission token, a user can set key
+value in the specified asset.
+
+### `CanRemoveKeyValueInUserAssets`
+
+With `CanRemoveKeyValueInUserAssets` permission token, a user can remove
+key value in the specified asset.
+
+### `CanSetKeyValueInUserMetadata`
+
+With `CanSetKeyValueInUserMetadata` permission token, a user can set key
+value in the metadata for the specified account.
+
+```rust
+let mouse_id = <Account as Identifiable>::Id::from_str("mouse@wonderland")?;
+
+// Create a new `CanSetKeyValueInUserMetadata` token that, when granted to another account,
+// allows it to set key value to the metadata in Mouse's account
+let permission_to_set_key_value_in_mouse_metadata: PermissionToken =
+    key_value::CanSetKeyValueInUserMetadata::new(mouse_id.clone()).into();
+```
+
+### `CanRemoveKeyValueInUserMetadata`
+
+With `CanRemoveKeyValueInUserMetadata` permission token, a user can remove
+key value in the metadata for the specified account.
+
+```rust
+let mouse_id = <Account as Identifiable>::Id::from_str("mouse@wonderland")?;
+
+// Create a new `CanRemoveKeyValueInUserMetadata` token that, when granted to another account,
+// allows it to remove key value from the metadata in Mouse's account
+let permission_to_remove_key_value_in_mouse_metadata: PermissionToken =
+    key_value::CanRemoveKeyValueInUserMetadata::new(mouse_id.clone()).into();
+```
+
+### `CanSetKeyValueInAssetDefinition`
+
+With `CanSetKeyValueInAssetDefinition` permission token, a user can set key
+value in the corresponding asset definition.
+
+### `CanRemoveKeyValueInAssetDefinition`
+
+With `CanRemoveKeyValueInAssetDefinition` permission token, a user can
+remove key value in the corresponding asset definition.
+
+### `CanRegisterDomains`
+
+with `CanRegisterDomains` permission token, a user can register domains.
+
+```rust
+let alice_id = AccountId::from_str("alice@test0")
+let mut alice = Account::new(alice_id.clone(), []).build();
+
+// Give Alice permission to register new domains
+alice.add_permission(register::CanRegisterDomains::new().into());
+```
```

Comment:
No-no, `Account::add_permission()` interface is used only inside Iroha 2 it-self. Please, use the aproach with `Grant` instruction, as in the other examples

Reply:
do I need to change the example for registering a new role as well? it also uses `add_permission`

```
let role_id = <Role as Identifiable>::Id::from_str("ACCESS_TO_MOUSE_METADATA")?;
let role = iroha_data_model::role::Role::new(role_id)
    .add_permission(CanSetKeyValueInUserMetadata::new(mouse_id))
    .add_permission(CanRemoveKeyValueInUserMetadata::new(mouse_id));
let register_role = RegisterBox::new(role);
```

Reply:
No, `Role::add_permission()` exists for public API. It just helps to construct `Role` object which then will be registered and granted. Without registration this role does not exists in Iroha 2.
But `Account::add_permission()` modifies the account localy and does not provide this information to Iroha, so it has no meaning for the external user (and we also exclude this from public API with a feature mechanism)